Training24/test.c

```diff
@@ -8,26 +8,117 @@
+   printf ("| Test Case | Input       | Expected     | Actual       |Result|\n");
+   printf ("|-----------|-------------|--------------|--------------|------|\n");
+   for (int i = 0; i < numTests; i++) {
+      // Adjust column widths to match header alignment
+      printf ("| %-9d | %-11d | %-12s | %-12s | %-6s |\n", i + 1, inputs[i], expected_results[i], actual_results[i] ? actual_results[i] : "NULL",
+         pass[i] ? COLOR_GREEN "PASS" COLOR_RESET : COLOR_RED "FAIL" COLOR_RESET);
+   }
+   printf ("\n");
+}
+
+/// <summary>Test cases for DecimalToBinary function.</summary>
+void TestDecToBinAndNegBin () {
+   // Define input test values and expected binary results
+   int inputs[] = { 5, 45, 214, 2147483647, 255, -5, -255, -2147483647 };
+   const char* expected_bin_result[] = {
+       "00000000 00000000 00000000 00000101",  // 5
+       "00000000 00000000 00000000 00101101",  // 45
+       "00000000 00000000 00000000 11010110",  // 214
+       "01111111 11111111 11111111 11111111",  // 2147483647
+       "00000000 00000000 00000000 11111111",  // 255
+       "11111111 11111111 11111111 11111011",  // -5
+       "11111111 11111111 11111111 00000001",  // -255
+       "10000000 00000000 00000000 00000001"   // -2147483647
+   };
+   char* actual_bin_result[sizeof (inputs) / sizeof (inputs[0])];                                    // Allocate arrays to store actual results and pass/fail status
+   int pass[sizeof (inputs) / sizeof (inputs[0])];
+   for (int i = 0; i < sizeof (inputs) / sizeof (inputs[0]); i++) {
+      char* result = DecimalToBinary (inputs[i]);
+      if (result && strcmp (result, expected_bin_result[i]) == 0) pass[i] = 1;                      // Pass
+      else pass[i] = 0;                                                                             // Fail
+      actual_bin_result[i] = result;
+   }
+   // Call the function to format and print the results
+   FormatBinary (sizeof (inputs) / sizeof (inputs[0]), inputs, expected_bin_result, actual_bin_result, pass);
+   for (int i = 0; i < sizeof (inputs) / sizeof (inputs[0]); i++) free (actual_bin_result[i]);      // Free allocated memory
+}
+
+/// <summary>Test cases for DecimalToHex function.</summary>
+void TestDecToHexAndNegHex () {
+   // Define input test values and expected hexadecimal results
+   int inputs[] = { 1, 255, 4095, 65536, 2147483647, -2147483647, -255, -1 };
+   const char* expected_hex_result[] = {
+       "00000001",  // 1
+       "000000FF",  // 255
+       "00000FFF",  // 4095
+       "00010000",  // 65536
+       "7FFFFFFF",  // 2147483647
+       "80000001",  // -2147483647
+       "FFFFFF01",  // -255
+       "FFFFFFFF"   // -1
+   };
+   char* actual_hex_result[sizeof (inputs) / sizeof (inputs[0])];                                 // Allocate arrays to store actual results and pass/fail status
+   int pass[sizeof (inputs) / sizeof (inputs[0])];

```

Comment:
**sizeof (inputs) / sizeof (inputs[0])**  this computation is used many places better store it in a variable and use